tests/integration/system/test_system_client.py

```diff
@@ -0,0 +1,259 @@
+from typing import List
+
+import pytest
+from nisystemlink.clients.core._http_configuration import HttpConfiguration
+from nisystemlink.clients.system import SystemClient
+from nisystemlink.clients.system.models import (
+    CancelJobRequest,
+    CreateJobRequest,
+    CreateJobResponse,
+    JobSummaryResponse,
+    QueryJobsRequest,
+)
+
+
+@pytest.fixture(scope="class")
+def client(enterprise_config: HttpConfiguration) -> SystemClient:
+    """Fixture to create an SystemClient instance."""
+    return SystemClient(enterprise_config)
+
+
+@pytest.fixture(scope="class")
+def create_job(
+    client: SystemClient,
+):
+    """Fixture to create a job."""
+    responses: List[CreateJobResponse] = []
+
+    def _create_job(job: CreateJobRequest) -> CreateJobResponse:
+        response = client.create_job(job)
+        responses.append(response)
+        return response
+
+    yield _create_job
+
+    from nisystemlink.clients.system.models import CancelJobRequest
+
+    job_requests = [
+        CancelJobRequest(jid=response.jid, tgt=response.tgt[0])
+        for response in responses
+        if response.tgt is not None
+    ]
+
+    client.cancel_jobs(job_requests)
+
+
+@pytest.fixture(scope="class")
+def create_multiple_jobs(
+    create_job,
+):
+    """Fixture to create multiple jobs."""
+    responses = []
+    arg_1 = [["A description"]]
+    arg_2 = [["Another description"]]
+    tgt = ["HVM_domU--SN-ec200972-eeca-062e-5bf5-33g3g3g3d73b2--MAC-0A-E1-20-D6-96-2B"]
+    fun_1 = ["system.set_computer_desc"]
+    fun_2 = ["system.set_computer_asc"]
+    metadata = {"queued": True, "refresh_minion_cache": {"grains": True}}
+    job_1 = CreateJobRequest(
+        arg=arg_1,
+        tgt=tgt,
+        fun=fun_1,
+        metadata=metadata,
+    )
+    responses.append(create_job(job_1))
+
+    job_2 = CreateJobRequest(
+        arg=arg_2,
+        tgt=tgt,
+        fun=fun_2,
+        metadata=metadata,
+    )
+    responses.append(create_job(job_2))
+
+    return responses
+
+
+@pytest.mark.integration
+@pytest.mark.enterprise
+class TestSystemClient:
+    def test__create_job__succeeds(
```

Comment:
can we update test names with proper details?

Reply:
I have updated the tests.